native/core/src/execution/operators/scan.rs

```diff
@@ -365,28 +378,23 @@ struct ScanStream<'a> {
     scan: ScanExec,
     /// Schema representing the data
     schema: SchemaRef,
-    /// Metrics
-    baseline_metrics: BaselineMetrics,
     /// Cast options
     cast_options: CastOptions<'a>,
     /// elapsed time for casting columns to different data types during scan
     cast_time: Time,
+    /// elapsed time for fetching arrow arrays from JVM
+    jvm_fetch_time: Time,
 }
 
 impl<'a> ScanStream<'a> {
-    pub fn new(
-        scan: ScanExec,
-        schema: SchemaRef,
-        partition: usize,
-        baseline_metrics: BaselineMetrics,
-    ) -> Self {
+    pub fn new(scan: ScanExec, schema: SchemaRef, partition: usize, jvm_fetch_time: Time) -> Self {
```

Comment:
perhaps `jvm_fetch_time` enough for now, but if you wanna expand metrics in future its better to have a wrapper structure similar to `BaselineMetrics` ?